tutorials/peft-curation/docbuilder.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+from typing import Set, Tuple
+
+import requests
+
+from nemo_curator.download.doc_builder import (
+    DocumentDownloader,
+    DocumentExtractor,
+    DocumentIterator,
+)
+
+
+class EmailsDownloader(DocumentDownloader):
```

Comment:
These look super clean, do you think you could plug them into the `download_and_extract` [function](https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/download/doc_builder.py#L151) that we have? It would be great if you could show that off too. It's technically a bit overkill for this, but still gives users a good perspective.

Reply:
Unfortunately not possible. There's an implicit assumption about the `text` field [here](https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/download/doc_builder.py#L133), which makes it unsuitable for this example.

Reply:
Very well. We'll need to update the logic then to make it more flexible in the future. For now, we'll just go with this.